rust/experimental/query_engine/engine-recordset/src/scalars/text_scalar_expressions.rs

```diff
@@ -51,6 +51,164 @@ where
                 || format!("Evaluated as: '{v}'"),
             );
 
+            Ok(v)
+        }
+        TextScalarExpression::Match(m) => {
+            // Execute the input expression to get the string to match
+            let input_value =
+                execute_scalar_expression(execution_context, m.get_input_expression())?;
+            let input_string = match input_value.to_value() {
+                Value::String(s) => s.get_value(),
+                _ => {
+                    return Err(ExpressionError::ValidationFailure(
+                        m.get_query_location().clone(),
+                        format!(
+                            "Text match input must be a string, got: '{:?}'",
+                            input_value.get_value_type()
+                        ),
+                    ));
+                }
+            };
+
+            // Execute the pattern expression to get the regex
+            let pattern_value = execute_scalar_expression(execution_context, m.get_pattern())?;
+            let pattern_string = match pattern_value.to_value() {
+                Value::String(s) => s.get_value(),
+                _ => {
+                    return Err(ExpressionError::ValidationFailure(
+                        m.get_query_location().clone(),
+                        format!(
+                            "Text match pattern must be a string, got: '{:?}'",
+                            pattern_value.get_value_type()
+                        ),
+                    ));
+                }
+            };
+
+            // Compile the regex pattern
+            let regex = match regex::Regex::new(pattern_string) {
+                Ok(regex) => regex,
+                Err(e) => {
+                    return Err(ExpressionError::ValidationFailure(
+                        m.get_query_location().clone(),
+                        format!("Invalid regex pattern '{}': {}", pattern_string, e),
+                    ));
+                }
+            };
+
+            // Check if the regex matches the input string
+            let matches = regex.is_match(input_string);
+
+            let v = ResolvedValue::Computed(OwnedValue::Boolean(BooleanValueStorage::new(matches)));
+
+            execution_context.add_diagnostic_if_enabled(
+                RecordSetEngineDiagnosticLevel::Verbose,
+                text_scalar_expression,
+                || format!("Evaluated as: '{v}'"),
+            );
+
+            Ok(v)
+        }
+        TextScalarExpression::Extract(e) => {
```

Comment:
Below I dropped some comments about the "try_resolve" methods. Here all the logic is implemented in the engine. But if you implement it in `Value` with some helpers you can share a lot of the code between the engine and expressions which makes the "try_resolve" methods easier to build. Math is a great example of this done well.